src/workflow/CommandExecutor.py

```
@@ -77,34 +77,85 @@
+        Args:
+            process: The subprocess.Popen object to stream from
+        """
+        def read_stdout():
+            """Read stdout in real-time"""
+            try:
+                for line in iter(process.stdout.readline, ''):
+                    if line:
+                        self.logger.log(line.rstrip(), 2)
+                    if process.poll() is not None:
+                        break
+            except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)